### PR TITLE
Bugfix MTE-3641 Use Xcode 15 for L10nBuild and L10nScreenshotTest

### PR DESCRIPTION
**Pull request by clarmso**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/MTE-3641)

## :bulb: Description
`fastlane` does not work with Xcode 16 at the moment. Let's stay in Xcode 15 just for the L10n screenshots.

Bitrise L10nBuild: https://app.bitrise.io/build/73949e8e-f035-4762-b4f3-9acaa00aed44?tab=log 🎉 

## :pencil: Checklist
You have to check all boxes before merging
- [ ] Filled in the above information (tickets numbers and description of your work)
- [ ] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

### CODE REVIEW

File: firefox-ios/firefox-ios-tests/Tests/L10nSnapshotTests/L10nBaseSnapshotTests.swift [hunk 1]
```diff
@@ -140,4 +140,11 @@ extension XCUIElement {
         L10nBaseSnapshotTests().mozWaitForElementToExist(self, timeout: timeout)
         self.tap()
     }
+
+    /// Waits for the UI element and then taps and types the provided text if it exists.
+    func tapAndTypeText(_ text: String, timeout: TimeInterval? = TIMEOUT) {
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/L10nSnapshotTests/L10nBaseSnapshotTests.swift:145]: Fix an existing compilation issue. A recent addition `tapAndTypeText` has been introduced in `BaseTestCase` but it is not propagated to `L10nBaseSnapshotTests` class.

File: firefox-ios/firefox-ios-tests/Tests/L10nSnapshotTests/L10nSuite1SnapshotTests.swift [hunk 2]
```diff
@@ -70,7 +70,7 @@ class L10nSuite1SnapshotTests: L10nBaseSnapshotTests {
 
         app.buttons["\(rootA11yId)PrimaryButton"].tap()
         currentScreen += 1
-        mozWaitForElementToExist(app.textFields["url"])
+        mozWaitForElementToExist(app.textFields[AccessibilityIdentifiers.Browser.AddressToolbar.searchTextField])
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/L10nSnapshotTests/L10nSuite1SnapshotTests.swift:73]: The identifier has been updated in a recent Firefox iOS build.

File: firefox-ios/firefox-ios-tests/Tests/L10nSnapshotTests/L10nSuite1SnapshotTests.swift [hunk 3]
```diff
@@ -188,7 +188,7 @@ class L10nSuite1SnapshotTests: L10nBaseSnapshotTests {
 
     @MainActor
     func testETPperSite() {
-        mozWaitForElementToExist(app.buttons[AccessibilityIdentifiers.Toolbar.settingsMenuButton], timeout: 10)
+        mozWaitForElementToExist(app.buttons[AccessibilityIdentifiers.Toolbar.settingsMenuButton])
```
Comment (clarmso) [inline firefox-ios/firefox-ios-tests/Tests/L10nSnapshotTests/L10nSuite1SnapshotTests.swift:191]: Some tests timed out intermittently. The default timeout of `mozWaitForElementToExist` is 15. Let's remove some shorter timeout as a workaround for slower app startup.

File: bitrise.yml [hunk 4]
```diff
@@ -843,6 +843,10 @@ workflows:
     description: >-
       This Workflow is to run L10n tests in one locale and then share the bundle with
       the rest of the builds
+    meta:
```
Comment (clarmso) [inline bitrise.yml:846]: Force L10n* workflows for use Xcode 15.4 (for now!)

**Review by mdotb-moz — APPROVED** [review]
LGTM

File: firefox-ios/firefox-ios-tests/Tests/L10nSnapshotTests/L10nSuite1SnapshotTests.swift [hunk 5]
```diff
@@ -200,7 +200,7 @@ class L10nSuite1SnapshotTests: L10nBaseSnapshotTests {
         // Website without blocked elements
         navigator.openURL(loremIpsumURL)
         mozWaitForElementToNotExist(app.staticTexts["XCUITests-Runner pasted from Fennec"])
-        mozWaitForElementToExist(app.buttons[AccessibilityIdentifiers.Toolbar.trackingProtection], timeout: 5)
+        // mozWaitForElementToExist(app.buttons[AccessibilityIdentifiers.Toolbar.trackingProtection])
```
Comment (isabelrios) [inline firefox-ios/firefox-ios-tests/Tests/L10nSnapshotTests/L10nSuite1SnapshotTests.swift:203]: Can we remove this?

File: firefox-ios/firefox-ios-tests/Tests/L10nSnapshotTests/L10nSuite1SnapshotTests.swift [hunk 6]
```diff
@@ -212,7 +212,7 @@ class L10nSuite1SnapshotTests: L10nBaseSnapshotTests {
 
     @MainActor
     func testSettingsETP() {
-        mozWaitForElementToExist(app.buttons[AccessibilityIdentifiers.Toolbar.settingsMenuButton], timeout: 10)
+        mozWaitForElementToExist(app.buttons[AccessibilityIdentifiers.Toolbar.settingsMenuButton])
```
Comment (isabelrios) [inline firefox-ios/firefox-ios-tests/Tests/L10nSnapshotTests/L10nSuite1SnapshotTests.swift:215]: wondering if removing all the timeouts could make the tests more flaky....we could use the variable TIMEOUT instead? if those are not needed, I am happy with their removal

**Review by isabelrios — APPROVED** [review]
Just two comments to address / clarify. Thanks!

File: firefox-ios/firefox-ios-tests/Tests/L10nSnapshotTests/L10nSuite1SnapshotTests.swift (hunk 5 above)
Comment (clarmso) [reply firefox-ios/firefox-ios-tests/Tests/L10nSnapshotTests/L10nSuite1SnapshotTests.swift:203]: Yes, we can remove this. The tracking protection icon does not necessarily appear on the toolbar.

**Pull request merged**